mmv1/products/documentaiwarehouse/DocumentSchema.yaml

```diff
@@ -0,0 +1,265 @@
+properties:
+  - !ruby/object:Api::Type::String
+    name: 'name'
+    output: true
+    description: |
+      The resource name of the document schema.
+  - !ruby/object:Api::Type::String
+    name: 'displayName'
+    description: |
+      Name of the schema given by the user.
+  - !ruby/object:Api::Type::Boolean
+    name: 'documentIsFolder'
+    description: |
+      Tells whether the document is a folder or a typical document.
+  - !ruby/object:Api::Type::Array
+    name: 'propertyDefinitions'
+    description: |
+      Defines the metadata for a schema property.
+    item_type: !ruby/object:Api::Type::NestedObject
+      properties:
+        - !ruby/object:Api::Type::String
+          name: 'name'
+          description: |
+            The name of the metadata property.
+        - !ruby/object:Api::Type::String
+          name: 'displayName'
+          required: true
+          description: |
+            The display-name for the property, used for front-end.
+        - !ruby/object:Api::Type::Boolean
+          name: 'isRepeatable'
+          description: |
+            Whether the property can have multiple values.
+        - !ruby/object:Api::Type::Boolean
+          name: 'isFilterable'
+          description: |
+            Whether the property can be filtered. If this is a sub-property, all the parent properties must be marked filterable.
+        - !ruby/object:Api::Type::Boolean
+          name: 'isSearchable'
+          description: |
+            Indicates that the property should be included in a global search.
+        - !ruby/object:Api::Type::Boolean
+          name: 'isMetadata'
+          description: |
+            Whether the property is user supplied metadata.
+        - !ruby/object:Api::Type::Boolean
+          name: 'isRequired'
+          description: |
+            Whether the property is mandatory.
+        - !ruby/object:Api::Type::Enum
+          name: 'retrievalImportance'
+          description: |
+            Stores the retrieval importance.
+          values:
+            - :HIGHEST
+            - :HIGHER
+            - :HIGH
+            - :MEDIUM
+            - :LOW
+            - :LOWEST
+          default_value: :MEDIUM
+        - !ruby/object:Api::Type::Array
+          name: 'schemaSources'
+          description: |
+            The schema source information.
+          item_type: !ruby/object:Api::Type::NestedObject
+            properties:
+              - !ruby/object:Api::Type::String
+                name: 'name'
+                description: |
+                  The schema name in the source.
+              - !ruby/object:Api::Type::String
+                name: 'processorType'
+                required: true
+                description: |
+                  The Doc AI processor type name.
+        # select any one of the property types (integer, float, text, property, enum, dateTime, map, timestamp)
+        - !ruby/object:Api::Type::NestedObject
+          name: 'integerTypeOptions'
```

Comment:
Should this object always be empty? Or should it be a map with just one key: value pair?

Reply:
Yes, it should always be empty, because integerTypeOptions, floatTypeOptions etc don't have any actual properties according to the API, but Terraform won't accept a NestedObject having no properties. It is a little hacky, but to me it seems like the only solution. Please recommend if something else can be done

Reply:
You can set properties to empty. `properties: []`

Example: https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/products/dataplex/Datascan.yaml#L196